geco/generator.py

```diff
@@ -17,3 +18,41 @@ def __next__(self):
 
     def seed(self, seed: int):
         self.rng.seed(seed)
+
+
+@py_random_state(-1)
+def common_substructure_generator(
+    instance_generation_function,
+    instance_params_generation_function,
+    base_params,
+    new_params,
+    expand_params_function,
+    seed=0,
+):
+    """
+    Generates instances that have common substructure
+
+    Parameters
+    ----------
+    instance_generation_function:
+        base function that defines MIP
+    instance_params_generation_function:
+        function to generate parameters for instance_generation_function
+    base_params:
+        parameters for the common substructure
+    new_params:
```

Comment:
What about also merging `base_params` and `expand_params_function` into `expand_params_function` only.
The actual `expand_params_function` could accept parameters for backbone extension.

Reply:
or what about making a partial function that takes only a seed 😄

Reply:
but it might be too difficult for users?

Reply:
That's what I head in mind. If one wants to have a specific extension function, let's say for set-packing that makes an (n,m) instance an (n',m') instance, one can just take the general extension function and specify it using partial / lambda.

Reply:
I implemented it in the test using lambda, can you check if this is what you had in mind?